server/src/main/java/com/hedera/block/server/mediator/LiveStreamMediatorImpl.java

```diff
@@ -0,0 +1,112 @@
+ * limitations under the License.
+ */
+
+package com.hedera.block.server.mediator;
+
+import com.hedera.block.protos.BlockStreamServiceGrpcProto;
+import com.hedera.block.server.consumer.LiveStreamObserver;
+import com.hedera.block.server.persistence.BlockPersistenceHandler;
+
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.logging.Logger;
+
+/**
+ * LiveStreamMediatorImpl is the implementation of the StreamMediator interface.  It is responsible for managing
+ * the subscription and unsubscription operations of downstream consumers.  It also proxies new blocks
+ * to the subscribers as they arrive and persists the blocks to the block persistence store.
+ *
+ */
+public class LiveStreamMediatorImpl implements StreamMediator<BlockStreamServiceGrpcProto.Block, BlockStreamServiceGrpcProto.BlockResponse> {
+
+    private final Logger LOGGER = Logger.getLogger(getClass().getName());
+    private final Set<LiveStreamObserver<BlockStreamServiceGrpcProto.Block, BlockStreamServiceGrpcProto.BlockResponse>> subscribers = Collections.synchronizedSet(new LinkedHashSet<>());
+
+    private final BlockPersistenceHandler<BlockStreamServiceGrpcProto.Block> blockPersistenceHandler;
+
+    /**
+     * Constructor for the LiveStreamMediatorImpl class.
+     *
+     * @param blockPersistenceHandler the block persistence handler
+     *
+     */
+    public LiveStreamMediatorImpl(final BlockPersistenceHandler<BlockStreamServiceGrpcProto.Block> blockPersistenceHandler) {
+        this.blockPersistenceHandler = blockPersistenceHandler;
+    }
+
+    /**
+     * Subscribe a new observer to the mediator
+     *
+     * @param liveStreamObserver - the observer to be subscribed
+     */
+    @Override
+    public void subscribe(final LiveStreamObserver<BlockStreamServiceGrpcProto.Block, BlockStreamServiceGrpcProto.BlockResponse> liveStreamObserver) {
+        this.subscribers.add(liveStreamObserver);
+    }
+
+    /**
+     * Unsubscribe an observer from the mediator
+     *
+     * @param liveStreamObserver - the observer to be unsubscribed
+     */
+    @Override
+    public void unsubscribe(final LiveStreamObserver<BlockStreamServiceGrpcProto.Block, BlockStreamServiceGrpcProto.BlockResponse> liveStreamObserver) {
+        if (this.subscribers.remove(liveStreamObserver)) {
+            LOGGER.finer("Successfully removed observer from subscription list");
+        } else {
+            LOGGER.finer("Failed to remove observer from subscription list");
+        }
+    }
+
+    /**
+     * Check if an observer is subscribed to the mediator
+     *
+     * @param observer - the observer to be checked
+     * @return true if the observer is subscribed, false otherwise
+     */
+    @Override
+    public boolean isSubscribed(final LiveStreamObserver<BlockStreamServiceGrpcProto.Block, BlockStreamServiceGrpcProto.BlockResponse> observer) {
+        return this.subscribers.contains(observer);
```

Comment:
Quite a few unnecessary uses of `this` in the file.  It's inconsistent (sometimes used, sometimes not) as well.  Ideally, don't use `this` unless necessary.

Reply:
fixed